TCAT/Controllers/RouteDetail+DrawerViewController.swift

```diff
@@ -250,78 +204,173 @@ class RouteDetailDrawerViewController: UIViewController, UITableViewDataSource,
+        if let drawer = self.parent as? RouteDetailViewController {
+            switch drawer.drawerPosition {
 
-        let lastCellIndexPath = IndexPath(row: tableView.numberOfRows(inSection: 0) - 1, section: 0)
-        var screenBottom = main.height
-        if #available(iOS 11.0, *) {
-            screenBottom -= view.safeAreaInsets.bottom
+            case .collapsed, .partiallyRevealed:
+                if selectedDirection != nil {
+                    drawer.setDrawerPosition(position: .collapsed, animated: true)
+                } else {
+                    drawer.setDrawerPosition(position: .open, animated: true)
+                }
+
+            case .open:
+                drawer.setDrawerPosition(position: .collapsed, animated: true)
+
+            default: break
+
+            }
         }
+    }
+}
 
-        // Calculate height of space between last cell and the bottom of the screen, also accounting for summary
-        var footerHeight = screenBottom - (tableView.cellForRow(at: lastCellIndexPath)?.frame.maxY ?? screenBottom) - summaryView.frame.height
-        footerHeight = expandedCell != nil ? 0 : footerHeight
+extension RouteDetailDrawerViewController: LargeDetailTableViewDelegate {
 
-        emptyFooterView.frame.size = CGSize(width: view.frame.width, height: footerHeight)
-        emptyFooterView.contentView.backgroundColor = Colors.white
-        emptyFooterView.layoutIfNeeded()
+    func collapseCells(indexPath: IndexPath) {
+        toggleCellExpansion(at: indexPath)
+    }
 
-        // Create Footer for No Data from Live Tracking Footer, if needed
+    func expandCells(indexPath: IndexPath) {
+        if justLoaded { summaryTapped() }
 
-        guard
-            let drawer = self.parent as? RouteDetailViewController,
-            let contentViewController = drawer.primaryContentViewController as? RouteDetailContentViewController
-            else {
-                return emptyFooterView
-        }
+        toggleCellExpansion(at: indexPath)
 
-        var message: String?
+        tableView.layoutIfNeeded()
+        tableView.layoutSubviews()
+    }
+}
 
-        if !contentViewController.noDataRouteList.isEmpty {
-            if contentViewController.noDataRouteList.count > 1 {
-                message = Constants.Banner.noLiveTrackingForRoutes
+// MARK: Pulley Delegate
+extension RouteDetailDrawerViewController: PulleyDrawerViewControllerDelegate {
+    func collapsedDrawerHeight(bottomSafeArea: CGFloat) -> CGFloat {
+        return bottomSafeArea + summaryView.frame.height
+    }
+
+    func partialRevealDrawerHeight(bottomSafeArea: CGFloat) -> CGFloat {
+        return main.height / 2
+    }
+
+    func drawerPositionDidChange(drawer: PulleyViewController, bottomSafeArea: CGFloat) {
+
+        justLoaded = false
+
+        // Center map on drawer change
+        if drawer.drawerPosition == .collapsed || drawer.drawerPosition == .partiallyRevealed {
```

Comment:
would this be cleaner as a `switch` with two cases instead? @kevinchan159

Reply:
+1